cmd/ir2proxy/main.go

```diff
@@ -38,31 +38,36 @@ func main() {
 		log.Fatal(err)
 	}
 
-	contourscheme.AddToScheme(scheme.Scheme)
-
+	// Split the yaml file by docs, and parse each one individually
 	for _, yamldoc := range bytes.Split(data, []byte("---")) {
 		if len(yamldoc) == 0 {
 			continue
 		}
 
-		decode := scheme.Codecs.UniversalDeserializer().Decode
-
-		ir, groupVersionKind, err := decode(yamldoc, nil, nil)
+		ir, err := k8sdecoder.DecodeIngressRoute(yamldoc)
 		if err != nil {
-			log.Infof("Skipped yaml doc, %s", err)
+			log.Error(err)
 			continue
 		}
-		switch t := ir.(type) {
-		case *irv1beta1.IngressRoute:
-			log.Infof("IngressRoute %s, namespace %s, labels %s", t.ObjectMeta.Name, t.ObjectMeta.Namespace, t.ObjectMeta.Labels)
-			hp, err := translateIngressRouteToHTTPProxy(t)
-			if err != nil {
-				log.Fatal(err)
+
+		validateWarnings := validate.CheckIngressRoute(ir)
+		if len(validateWarnings) > 0 {
```

Comment:
if the length is zero then the loop will execute zero times

Reply:
True, but as you said, the program should then exit, so we need this if, sadly.